testsuite/simple/hibernate-search/src/test/java/io/thorntail/testsuite/hibernate_search/HibernateSearchAppTest.java

```diff
@@ -0,0 +1,22 @@
+package io.thorntail.testsuite.hibernate_search;
+
+import io.thorntail.test.ThorntailTestRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.hibernate.search.jpa.FullTextEntityManager;
+
+@RunWith(ThorntailTestRunner.class)
+public class HibernateSearchAppTest {
+    @PersistenceContext
+    EntityManager em;
```

Comment:
Can also test that it's possible to do `@PersistenceContext FullTextEntityManager em;`?

Reply:
I looked into that quite extensively and from what I can tell it would involve a re-write of some of Weld's internal classes.

I can't remember which file it's in, but I think the line is something like `Class.forName("javax.persistence.PersistenceContext");` in the `weld/api` repo. I am not certain it can't be done, but it appears to be very difficult. I'd love to get a Weld expert to weigh in on this. 👍

Reply:
Does the Hibernate ORM EntityManager#unwrap method know how to obtain the FullTextEntityManager?

Reply:
Not from what I saw in the Hibernate search docs, but that would be really nice. I will try that later today when I get to a computer.

Reply:
@scottmarlow It looks like the answer is no, unwrap does not understand that. It makes sense since ORM doesn't depend on Search.